Ntuplizer/plugins/JpsiMuNtuplizer.cc

```diff
@@ -317,400 +318,478 @@ particle_cand JpsiMuNtuplizer::calculateIPvariables(
+            if(trigNames.triggerName(i).find("HLT_DoubleMu4_JpsiTrk_Displaced_v")!= std::string::npos){
+                nBranches_->HLT_BPH_isFired[trigNames.triggerName(i)] = HLTtriggers_->accept(i);
+                if(HLTtriggers_->accept(i)){
+                    isTriggered = true;
+                    finalTriggerName=trigNames.triggerName(i);  
+                    finalTriggerFilterObjName="hltVertexmumuFilterJpsiMuon3p5";
+                    // std::cout << "finalTriggerName = "  << finalTriggerName << std::endl;
+                    
+                }
+            }
+
+        }
+    }
 
-    const pat::Muon & muon = (*muons_)[imuon];
 
-    if(muon.pt() < 4) continue;
-    if(TMath::Abs(muon.eta()) > 2.4) continue;
-    if(!(muon.track().isNonnull())) continue;
 
-    //    bool isSoft = muon.isSoftMuon(*firstGoodVertex);
-    //    bool isGlobal = muon.isGlobalMuon();
-    //    bool isTracker = muon.isTrackerMuon();
-    //    bool isLoose = muon.isLooseMuon();
-    //    bool isTight =  muon.isTightMuon(*firstGoodVertex);
-    //    bool isPF = muon.isPFMuon();
-    //    if(!(isSoft && isGlobal)) continue;
-    //    if(TMath::Abs(muon.muonBestTrack()->dz(firstGoodVertex->position())) > 0.5) continue;
+    /********************************************************************
+     *
+     * Step2: pre-select muons for building J/psi candidates ... 
+     * For muons, no requirement applied
+     *
+     ********************************************************************/
+
+    event.getByToken(verticeToken_   , vertices_     );
+    event.getByToken(bsToken_   , beamspot_     );
+    event.getByToken(muonToken_	, muons_    );
+    event.getByToken(triggerObjects_  , triggerObjects);
+
+    std::vector<pat::Muon> muoncollection;
+    std::vector<int> muoncollection_id;
+    muoncollection.clear();
+    muoncollection_id.clear();
+
+    // bools keeping track of muon 1 or 2 for cutflow purposes
+    bool mu1passpteta = false;
+    bool mu2passpteta = false;
+    bool mu3passpteta = false;
+    bool mu1passtrigmatch = false;
+    bool mu2passtrigmatch = false;
+ // Precut
+    if( doCutFlow_) {
+       nBranches_->cutflow_perevt->Fill(0);
+      
+       for(size_t imuon = 0; imuon < muons_->size(); ++ imuon){
+
+           const pat::Muon & muon = (*muons_)[imuon];
+           // right now we're just checking that there are at least two muons passing our kinematic cuts
+           // so we don't care if there are more than two muons that pass right now
+           if(mu2passpteta) continue;
+           if(muon.pt() < 4) continue;
+           if(TMath::Abs(muon.eta()) > 2.4) continue;
+           if(!(muon.track().isNonnull())) continue;
+           // If we're on the Second muon, then the second muon is what passed 
+           if (mu1passpteta && !mu2passpteta) { mu2passpteta = true;}
+           // If we're on the First muon, then the first muon is what passed
+           if (!mu1passpteta) {mu1passpteta = true;}
```

Comment:
I don't understand what do you want to do here ...

Reply:
I'll let Wren comment this one, since she implementted it. But I guess she was just checking the fact that there are 2 muons in the sample passing the kinemtic selection.

Reply:
Hmmm ... but you have almost duplicate of the codes between L397 to L488, which I don't like. Probably, my question is, rather, why do you need cutflow for each of the muon, rather than just J/psi selection ? Why do we need such granularity?

Reply:
@yuta, this was just to check the basic kinematic selection efficiency on the samples that I produced to check the validity of the samples and find out if the gen filters cut where appropriately placed. In further iterations of the code we could remove this bit and have just the J/psi selection.

Reply:
ok got it.